packages/client/hmi-client/src/components/workflow/ops/model/tera-model-node.vue

```diff
@@ -73,14 +73,17 @@ async function getModelById(modelId: string) {
 	model.value = await getModel(modelId);
 
 	if (model.value && model.value.id) {
-		const state = _.cloneDeep(props.node.state);
-		state.modelId = model.value?.id;
-		emit('update-state', state);
-		emit('append-output', {
-			type: 'modelId',
-			label: model.value.header.name,
-			value: [model.value.id]
-		});
+		const outputs = props.node.outputs;
+		if (_.isEmpty(outputs) || (outputs.length === 1 && !outputs[0].value)) {
```

Comment:
Since this check is kinda elaborate and repeated maybe a helper function like `shouldAppendDefaultOutput` in `services/workflow.ts` could help clean things up.